src/common/quantile.cc

```diff
@@ -283,7 +285,9 @@ void SketchContainerImpl<WQSketch>::AllReduce(
   collective::Allreduce<collective::Operation::kMax>(&n_columns, 1);
   CHECK_EQ(n_columns, sketches_.size()) << "Number of columns differs across workers";
 
-  AllreduceCategories(feature_types_, n_threads_, &categories_);
+  if (!col_split_) {
```

Comment:
Maybe we should `LOG(FATAL)` in all reduce categories if this is a column split matrix?

Reply:
`AllreduceCategories` is a freestanding function that doesn't have access to `this->col_split_`. Maybe we should make it a member function?

Reply:
Made it a private member function. PTAL